arcgis-ios-sdk-samples/Features/Display feature layers/DisplayFeatureLayersViewController.swift

```diff
@@ -0,0 +1,189 @@
+            } else {
+                // Get the feature table with the name.
+                let featureTable = self.geodatabase.geodatabaseFeatureTable(withName: "Trailheads")!
+                // Create a feature layer with the feature table.
+                let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+                // Set the viewpoint to Malibu, California.
+                let viewpoint = AGSViewpoint(center: AGSPoint(x: -13214155, y: 4040194, spatialReference: .webMercator()), scale: 35e4)
+                self.setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+            }
+        }
+    }
+    
+    /// Load a feature layer with a local geopackage.
+    func loadGeopackage() {
+        // Create a geopackage from a named bundle resource.
+        geoPackage = AGSGeoPackage(name: "AuroraCO")
+        
+        // Load the geopackage.
+        geoPackage?.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            }
+            // Add the first feature layer from the geopackage to the map.
+            if let featureTable = self.geoPackage?.geoPackageFeatureTables.first {
+                // Create the feature layer with the feature table.
+                let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+                // Set the viewpoint to Aurora, Colorado.
+                let viewpoint = AGSViewpoint(latitude: 39.7294, longitude: -104.8319, scale: 577790.554289)
+                self.setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+            }
+        }
+    }
+    
+    /// Load a feature layer with a local shapefile.
+    func loadShapefile() {
+        // Create a shapefile feature table from a named bundle resource.
+        let shapefileTable = AGSShapefileFeatureTable(name: "ScottishWildlifeTrust_ReserveBoundaries_20201102")
+        
+        // Create a feature layer for the shapefile feature table.
+        let featureLayer = AGSFeatureLayer(featureTable: shapefileTable)
+        // Ensure the feature layer's metadata is loaded.
+        featureLayer.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            }
+            // Set the viewpoint to Scotland.
+            let viewpoint = AGSViewpoint(latitude: 56.641344, longitude: -3.889066, scale: 6e6)
+            self.setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+        }
+    }
+    
+    /// Add the feature layer to the map and set the viewpoint
+    /// featureLayer - The feature layer to display and add to the map.
+    /// viewpoint - The viewpoint to change the map to.
+    func setMap(featureLayer: AGSFeatureLayer, viewpoint: AGSViewpoint) {
+        mapView.map?.operationalLayers.removeAllObjects()
+        mapView.map?.operationalLayers.add(featureLayer)
```

Comment:
I wonder if Phil would suggest `setArray` method here; what you have is fine.

Reply:
I'll leave this conversation open

Reply:
Yes, using a single operation (`setArray`) is preferred to multiple operations (`removeAllObjects` + `add`).